modules/dokkatoo-plugin/src/main/kotlin/DokkatooExtension.kt

```diff
@@ -87,17 +87,32 @@ constructor(
    *
    * Dokka will merge Dokka Source Sets from other subprojects if...
    */
-  abstract val dokkatooSourceSets: NamedDomainObjectContainer<DokkaSourceSetSpec>
+  val dokkatooSourceSets: NamedDomainObjectContainer<DokkaSourceSetSpec> =
+    extensions.adding("dokkatooSourceSets") {
+      objects.domainObjectContainer()
+    }
 
   /**
    * Dokka Plugin are used to configure the way Dokka generates a format.
    * Some plugins can be configured via parameters, and those parameters are stored in this
    * container.
    */
   val pluginsConfiguration: DokkaPluginParametersContainer =
-    objects.polymorphicDomainObjectContainer(DokkaPluginParametersBaseSpec::class)
+    extensions.adding("pluginsConfiguration") {
+      objects.dokkaPluginParametersContainer()
+    }
+
+  /**
+   * Versions of dependencies that Dokkatoo will use to run Dokka Generator.
+   *
+   * These versions can be set to change the versions of dependencies that Dokkatoo uses defaults,
+   * or can be read to align versions.
+   */
+  val versions: Versions = extensions.adding("versions") {
+    objects.newInstance()
+  }
```

Comment:
@martinbonnin 👋 this change should help with your comment https://github.com/adamko-dev/dokkatoo/pull/62#discussion_r1183583433

Reply:
Thanks! Out of curiosity, why do you need `versions` to be an extension as opposed to a regular JVM property (`val versions: Versions`) ? 

Since it's always going to be there, there's no need to generate accessors at runtime? Or is it?

Reply:
if it's an extension then Gradle will generate a nice accessor, so you can do 

```kotlin
dokkatoo {
  versions {
    ...
  }
}
```

Reply:
You could always add it manually?

```kotlin
class DokkatooExtension {
  val versions: Versions = VersionsImpl()

  fun versions(action: Action<Version>) {
     action.execute(versions)
  }
}
```

I don't have autogenerated accessors in my convention plugins so I tend to prefer explicit but either works, was mainly curious

Reply:
Yeah, for sure, an explicit function works too. My reasoning was that I wanted to try and use as many Gradle features as possible. Also, to try and keep the API footprint small, and since there are scope functions (so `versions.apply {}` is possible) is a specific function worth it? I'm on the fence...

Reply:
> I'm team "Gradle has too many features/ways to do things and generated accessors take time to generate".

Good point! I didn't consider that angle.

> Other philosophical question: is there a use case for the user to actually set `jetbrainsDokka`? Or could `jetbrainsDokka` be a simple `val` because changing it to anything else is risking an incompatibility between dokkatoo and dokka?

Yeah, it risks incompatibility, but only if Dokka's ABI changes. Dokkatoo runs Dokka Generator using an isolated worker, so it's possible (in theory) to keep using the same version of Dokkatoo and bump or rollback the Dokka version, or vice versa. 

For example, if I've got a project set up and I want to use some new version of Gradle, and for some compatibility reason Dokkatoo needs to be updated too. However, I want to keep Kotlin and Dokka at the same version. Rather than being forced to update Dokkatoo and Dokka at the same time, I can override the Dokka version, and my application and documentation should be exactly the same.